src/sagemaker/model.py

```diff
@@ -178,21 +178,23 @@ def register(
         """
         if self.model_data is None:
             raise ValueError("SageMaker Model Package cannot be created without model data.")
+        if image_uri is not None:
+            self.image_uri = image_uri
+        container_def = self.prepare_container_def()
 
         model_pkg_args = sagemaker.get_model_package_args(
             content_types,
             response_types,
             inference_instances,
             transform_instances,
-            model_package_name,
-            model_package_group_name,
-            self.model_data,
```

Comment:
Is it intentional to remove this argument? I think we need this.

Reply:
Actually they are not removed, but passed in as named parameters. This is required as some of the parameters are optional. Also the model_data and image_url are now packed within container_def.

Reply:
ok thanks, didn't know `container_def` makes use of model_data and image_uri.